[New feature] multiple workflows (to switch between workflows)

TomoyukiMizuma opened the pull request:
I added a new button "Workflow".
You can switch between workflows by clicking the button.
![new](https://github.com/comfyanonymous/ComfyUI/assets/53853234/8046edbe-6d5f-4809-80ea-2e7e7a0cd8f2)
related PR [#310](url:https://github.com/comfyanonymous/ComfyUI/pull/310)

ltdrdata commented:
How about combo other than button.

TomoyukiMizuma commented:
Updated. The combo version instead of the button.
![new](https://github.com/comfyanonymous/ComfyUI/assets/53853234/e1d3d3e0-eb82-422f-8fcf-8dc9c95a4c8d)

ltdrdata commented:
That's good. How about that providing settings for workflow count on settings menu.

TomoyukiMizuma commented:
Updated. I added a new feature (Rename workflows dialog).
I added an open dialog button to the settings menu.
![new1](https://github.com/comfyanonymous/ComfyUI/assets/53853234/596bc6da-2959-4266-a938-85ce44e09a9b)
I didn't add dynamically changing the count of workflows while the app is running because it would complicate the code.

TomoyukiMizuma reviewed (commented):
I added a new feature to seamlessly switch between workflows.
I also added a new button to rename workflows in the settings menu.

2blackbar commented:
finally some sanity

dimtoneff commented:
OMG I NEED THIS!

comfyanonymous commented:
Is there a reason why the button to change workflows is locked when it's generating?

M1kep commented on web/scripts/app.js:
```diff
@@ -1597,6 +1601,44 @@ export class ComfyApp {
 		this.runningNodeId = null;
 	}
+
+	switchWorkflow(seleted_id) {
```
`seleted_id` -> `selected_id`

M1kep commented:
What's the reasoning behind hardcoded count of workflows, vs an add/remove setup?

comfyanonymous commented:
Can you disable the workflow locking? I find it annoying especially when I open a new browser tab and want to work on another workflow.

comfyanonymous commented on web/scripts/app.js:
```diff
@@ -1600,6 +1609,44 @@ export class ComfyApp {
+
+	switchWorkflow(selected_id) {
+		// If you change the value of this workflow_count, you also need to change the count of `$el("option"~` (ui.js). (count up only)
+		const workflow_count = 10; // also used in getWorkflowNames function
```
You should only put this constant in one place and everything should depend on it.
